homeassistant/components/roku/config_flow.py

```diff
@@ -97,15 +95,24 @@ async def async_step_ssdp(
         """Handle a flow initialized by discovery."""
         host = urlparse(discovery_info[ATTR_SSDP_LOCATION]).hostname
         name = discovery_info[ATTR_UPNP_FRIENDLY_NAME]
-        serial_num = discovery_info[ATTR_UPNP_SERIAL]
+        serial_number = discovery_info[ATTR_UPNP_SERIAL]
 
-        await self.async_set_unique_id(serial_num)
+        await self.async_set_unique_id(serial_number)
         self._abort_if_unique_id_configured(updates={CONF_HOST: host})
 
         # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
-        self.context.update(
-            {CONF_HOST: host, CONF_NAME: name, "title_placeholders": {"name": host}}
-        )
+        self.context.update({"title_placeholders": {"name": name}})
+
+        self.discovery_info.update({CONF_HOST: host, CONF_NAME: name})
+
+        try:
```

Comment:
Can you give some background on why this was moved from `async_step_ssdp_confirm`?  It seems like it would better to do the I/O only after they confirm they want to set it up.

Reply:
i believe i did this based on feedback on another integration where Paulus didnt want something to be offered to user if it could fail so it was better to test before notification to user that triggers the confirm flow